price-estimator/src/main.rs

```diff
@@ -87,13 +79,25 @@ fn main() {
         .unwrap();
 
     let orderbook_task = runtime.spawn(update_orderbook_forever(
-        orderbook,
+        orderbook.clone(),
         options.orderbook_update_interval,
     ));
 
-    // TODO: handle http requests
-    let serve_task = runtime.spawn(warp::serve(warp::any().map(|| "")).run(([127, 0, 0, 1], 8080)));
+    let filter = filter::estimated_buy_amount().and_then(
+        move |token_pair: TokenPair, sell_amount_in_quote| {
+            let orderbook = orderbook.clone();
+            async move {
+                // TODO: format the response the way the nodejs price estimator did.
+                let orderbook = orderbook.get_reduced_orderbook().await;
+                let result = estimate_price(token_pair, sell_amount_in_quote, orderbook);
```

Comment:
We still need to calculate the buy amount after getting the price but I'll do that in the next PR.